derive-input/src/internal/integration.rs

```diff
@@ -29,21 +29,73 @@ fn get_table_attribute_macro(input: &DeriveInput) -> syn::Result<TokenStream> {
                         .to_string()
                         .eq("spacetimedb :: table")
                 {
-                    table = Some(list.tokens.clone());
+                    table_attrs.push(list.tokens.clone());
                 }
             }
             Err(_) => {}
         }
     }
 
-    match table {
-        Some(table) => Ok(table),
-        None => Err(Error::new(
-            // TODO: span should be the dsl attribute macro
+    if table_attrs.is_empty() {
+        return Err(Error::new(
             Span::call_site(),
             format!(
                 "Haven't found `#[table]`/`#[spacetimedb::table]` attribute macro! Make sure `#[dsl]`/`#[spacetimedsl::dsl]` is directly above one."
             ),
-        )),
+        ));
+    }
+
+    // If only one table, return it  
+    if table_attrs.len() == 1 {
+        return Ok(table_attrs[0].clone());
+    }
+
+    // For multiple table attributes, match each DSL macro with the appropriate table
+    // based on the naming convention in the DSL arguments (e.g., plural_name)
+    if let Some(plural_name) = extract_plural_name(&dsl_args.to_string()) {
+        for table_attr in &table_attrs {
+            let table_str = table_attr.to_string();
+            let base_name = if plural_name.contains("tables") {
+                plural_name.replace("tables", "table")
+            } else if plural_name.ends_with("s") && plural_name.len() > 1 {
+                plural_name[..plural_name.len()-1].to_string()
+            } else {
+                plural_name.clone()
+            };
+            
+            if table_str.contains(&base_name) {
+                return Ok(table_attr.clone());
+            }
+        }
+    }
+
+    // Fallback: use deterministic selection based on dsl args to ensure
+    // different DSL macros select different tables consistently
+    let selection_index = simple_hash(&dsl_args.to_string()) % table_attrs.len();
+    Ok(table_attrs[selection_index].clone())
+}
+
+// Extract plural_name from DSL arguments
+fn extract_plural_name(dsl_args: &str) -> Option<String> {
+    // Look for pattern like "plural_name = some_name"
+    if let Some(start) = dsl_args.find("plural_name") {
+        if let Some(equals_pos) = dsl_args[start..].find('=') {
+            let after_equals = &dsl_args[start + equals_pos + 1..];
+            // Find the next word/identifier, more carefully
+            let trimmed = after_equals.trim();
+            // Split by whitespace and commas to find the identifier
+            for part in trimmed.split(|c: char| c.is_whitespace() || c == ',') {
+                let cleaned = part.trim();
+                if !cleaned.is_empty() && cleaned.chars().all(|c| c.is_alphanumeric() || c == '_') {
+                    return Some(cleaned.to_string());
+                }
+            }
+        }
     }
+    None
+}
```

Comment:
Please use the established parsing logic of the spacetime-bindings-macro-input crate instead.


Good starting points to understand it's usage are the following files:

https://github.com/tamaro-skaljic/SpacetimeDSL/blob/main/derive-input/src/internal/dsl/table.rs
https://github.com/tamaro-skaljic/spacetime-bindings-macro-input/blob/main/src/table.rs